src/Plugin.php

```diff
@@ -40,16 +35,23 @@ class Plugin {
    * @implements init
    */
   public static function init() {
-    // Only users with an allowed role ('sale-editor' by default) or with the
-    // 'edit_product' capability can print the price label from the frontend.
-    if (!static::checkUserCanPrintLabel()) {
-      return;
+    // Set roles allowed to print products price labels.
+    // This constant can be set in wp-config.php file.
+    if (!defined('ROLES_CAN_PRINT_PRICE_LABEL')) {
+      define('ROLES_CAN_PRINT_PRICE_LABEL', ['sale-editor']);
     }
 
-    if (!Label::$pdfDocsFolderPath || !Label::$pdfDocsFolderUrl) {
-      $uploads_dir = wp_get_upload_dir();
-      Label::$pdfDocsFolderPath = $uploads_dir['basedir'] . '/' . Plugin::PREFIX;
-      Label::$pdfDocsFolderUrl = $uploads_dir['baseurl'] . '/' . Plugin::PREFIX;
+    // Add print price label capability to allowed roles.
+    static::addPrintLabelCapability(
```

Comment:
This will run the function on every page request. This method should be moved into the plugin activation/uninstall hook instead.

Reply:
You are so right! 🙈